compiler/codegen/OMRCodeGenerator.cpp

```diff
@@ -391,6 +400,47 @@ void OMR::CodeGenerator::lowerTrees()
       TR_ASSERT(node->getVisitCount() != visitCount, "Code Gen: error in lowering trees");
       TR_ASSERT(node->getReferenceCount() == 0, "Code Gen: error in lowering trees");
 
+      if (node->getOpCodeValue() == TR::BBStart)
```

Comment:
When this code was open sourced, we simplified and cleaned up the lowerTrees walk by providing "hooks" for the various operations people needed to do (the code was almost unreadable otherwise).

Please move this code under `lowerTreesPreTreeTopVisit()` or another appropriate hook.

Reply:
Yes, taking this code out will look better, but it will require another pass through the trees. This is because we maintain a few variables across the loop (eg. firstColdBlock). Having a separate function that handles one tree at a time and takes all those variables as reference parameters is an option. Please let me know which one you think is better. I feel like a separate pass would produce a cleaner code.

Reply:
Created new function.